waspc/examples/todoApp/ext/Todo.js

```diff
@@ -98,12 +99,25 @@ const Tasks = (props) => {
 }
 
 const Task = (props) => {
+  const action = useAction(updateTaskIsDone, {
+    optimisticUpdates: [{
+      getQuery: () => [getTasks],
```

Comment:
Is this an API that we define, or is it react query thing? What does getQuery stand for?

Reply:
It's an API we define.

#### The Jist
Ideally, it would only be `query` and we'd provide it with `[getTasks]`<sup>1</sup>.
Unfortunately, optimistic updates often require local lexical context for identifying the query (e.g., when updating a task with a specific ID). To support this, we need to have closures in the query-identifier array, and for that we need a function. To make it clear that we expect a function that returns a query (instead of directly expecting a query), we call the field `getQuery`.

#### The Long Story
A richer API could look something like this:
```javascript
const taskId = // we have access to the optimistically updated task ID here.
optimisticUpdates: [{
    query: [getTasks],
    // ...
  },
  {
    getQuery: () => [getTask, taskId],
    // ...
  }
]
```
Or we could have gone the usual javascript/python route and just treated the field differently depending on its type:
```javascript
const taskId = // we have access to the optimistically updated task ID here.
optimisticUpdates: [{
    query: [getTasks],
    // ...
  },
  {
    query: () => [getTasks, taskId],
    // ...
  }
]
```
I wasn't a fan of either approach, so I decided to limit the public API to only accept functions (as they're expressive enough to cover all use cases). And, since we're now only accepting functions, `getQuery` seemed like a more appropriate name:
```javascript
const taskId = // we have access to the optimistically updated task ID here.
optimisticUpdates: [{
    getQuery: () => [getTasks],
    // ...
  },
  {
    getQuery: () => [getTasks, taskId],
    // ...
  }
]
```
#### How's Any of This a "Query" anyway?
Finally, the last point of confusion probably comes from the implication that the array `[getTasks, taskId]` is a "query", which is different from all other parts of Wasp (where we consider the word "query" to mean a function). So, yes, I should probably rename that :)

React query calls a similar array a "query key", but that doesn't feel quite right. Their query key is `[queryCacheKey : string, ...otherStuff]`, while ours is `[queryFn: function, ...otherStuff]` (we decided to keep hiding the `queryCacheKey`, more on that [here](https://github.com/wasp-lang/wasp/pull/672#discussion_r923520647)). Calling them the same might cause confusion. Any ideas on how to approach this?

[1]  Technically, it would probably only be `getTasks` (without the array), but that's a [different story](https://github.com/wasp-lang/wasp/pull/672#discussion_r923518439). I'll assume we settled on always using the array.

Reply:
Ok all makes sense!
Regarding name -> well, probably best to do the usual practice of trying to explain it to somebody. So how would you explain to me, in sentence or couple of them, what is this thing, if not `query`? And then let's keep reducing that until we get an identifier/name that we can use.

Reply:
I'd call it a `querySpecifier`.

Reply:
Ha ok, specifier is very generic though :D. But it does make some sense and I don't really have a better idea

Reply:
Couldn't come up with anything better so I left it at specifier.